src/TGConnection/Socket/SocketAsyncTg.php

```diff
@@ -65,10 +65,10 @@ public function poll(): void
 
         try{
             $this->step->checkIfStepStuck();
-        } catch (SocksException $e){
+        } catch (Exception $e){
             $this->stop();
 
-            throw $e;
+            throw new SocksException(SocksException::STEP_STUCK, $e->getMessage());
```

Comment:
Why not wrap into TGException?

Reply:
It is wrapped in `ProxySocket`: https://github.com/Postuf/telegram-osint-lib/blob/master/src/TGConnection/Socket/ProxySocket.php#L177 , clients don't use `SocketAsyncTg` directly.
Marked this class as internal to emphasize this.